browser/ai_chat/ai_chat_settings_helper.cc

```diff
@@ -103,10 +134,68 @@ void AIChatSettingsHelper::GetManageUrl(GetManageUrlCallback callback) {
 #if defined(OFFICIAL_BUILD)
   std::string domain = brave_domains::GetServicesDomain(kAccountHostnamePart);
 #else
-  std::string domain = brave_domains::GetServicesDomain(kAccountHostnamePart, brave_domains::STAGING);
+  std::string domain = brave_domains::GetServicesDomain(kAccountHostnamePart,
+                                                        brave_domains::STAGING);
 #endif
 
-  std::move(callback).Run(base::StrCat({url::kHttpsScheme, url::kStandardSchemeSeparator, domain}));
+  std::move(callback).Run(
+      base::StrCat({url::kHttpsScheme, url::kStandardSchemeSeparator, domain}));
+}
+
+void AIChatSettingsHelper::GetCustomModels(GetCustomModelsCallback callback) {
+  std::vector<mojom::CustomModelPtr> custom_models = GetCustomModels();
+  std::move(callback).Run(std::move(custom_models));
+}
+
+void AIChatSettingsHelper::AddCustomModel(mojom::CustomModelPtr model,
+                                          AddCustomModelCallback callback) {
+  if (!IsEndpointUrlValid(model->endpoint)) {
+    std::move(callback).Run(mojom::OperationResult::InvalidUrl);
+    return;
+  }
+
+  model_service_->AddCustomModel(std::move(model));
+  std::move(callback).Run(mojom::OperationResult::Success);
+}
+
+void AIChatSettingsHelper::SaveCustomModel(uint32_t index,
+                                           mojom::CustomModelPtr model,
+                                           SaveCustomModelCallback callback) {
+  if (!IsEndpointUrlValid(model->endpoint)) {
+    std::move(callback).Run(mojom::OperationResult::InvalidUrl);
+    return;
+  }
+
```

Comment:
if the model host is `localhost` or resolving to localhost, we should replace with `127.0.0.1`

Reply:
That's not universal though, is it? As far as I understand, web servers could be "listening" to localhost but not 127.0.0.1 via host header

Reply:
It should be generic enough. After a discussion with @diracdeltas , it looks like this might be unrequired.
I would like to review this thing dynamically, before having a full resolution here